src/lib.rs.in

```diff
@@ -140,31 +142,43 @@ macro_rules! es_body_op {
 
 impl Client {
     /// Create a new client
-    pub fn new(host: &str, port: u32) -> Client {
+    pub fn new(host: &str) -> Client {
+        let mut url = Url::parse(host).unwrap();
+
+        if url.port().is_none() {
```

Comment:
Yeah, I think I agree with this, we might as well go all the way and require a full URL.  `https://someserver:9200/` etc.